storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogSegmentMetadata.java

```diff
@@ -100,7 +100,16 @@ public RemoteLogSegmentMetadata(RemoteLogSegmentId remoteLogSegmentId,
         this.remoteLogSegmentId = Objects.requireNonNull(remoteLogSegmentId, "remoteLogSegmentId can not be null");
         this.state = Objects.requireNonNull(state, "state can not be null");
 
+        if (startOffset < 0) {
+            throw new IllegalArgumentException(
+                String.format("Unexpected start offset = %d. StartOffset for a remote segment cannot negative", startOffset));
```

Comment:
typo: "remote segment cannot negative" -> "remote segment can not be negative"

Reply:
fixed